backend/open_webui/routers/functions.py

```
@@ -42,6 +51,111 @@ async def get_functions(user=Depends(get_admin_user)):
+############################
+# SyncFunctions
+############################
+
+
+class SyncFunctionsForm(BaseModel):
+    functions: list[FunctionModel] = []
+
+
+@router.post("/sync", response_model=list[FunctionModel])
+async def sync_functions(
+    request: Request, form_data: SyncFunctionsForm, user=Depends(get_admin_user)
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)